rclpy/rclpy/qos_event.py

```diff
@@ -94,6 +108,20 @@ class QoSSubscriptionEventType(IntEnum):
         ('total_count_change', 'int'),
     ])
 
+"""
+Payload type for Publisher Incompatible QoS callback.
+
+Mirrors rmw_offered_incompatible_qos_status_t from rmw/types.h
+"""
+QoSOfferedIncompatibleQoSInfo = QoSRequestedIncompatibleQoSInfo
+
+
+class UnsupportedEventTypeException(Exception):
```

Comment:
`RuntimeError` sounds like a better base class.
I would also rename the exception to `UnsupportedEventTypeError`, as `Error` is usually used as suffix in Python instead of `Exception`.

Reply:
Actually, `NotImplementedError` may be a better base class.

Reply:
The [Python documentation](https://docs.python.org/3/library/exceptions.html) says "All user-defined exceptions should also be derived from this class (Exception)", so I don't think `RuntimeError` or `NotImplementedError` are meant to be base classes.

Reply:
> The [Python documentation](https://docs.python.org/3/library/exceptions.html) says "All user-defined exceptions should also be derived from this class (Exception)", so I don't think `RuntimeError` or `NotImplementedError` are meant to be base classes.

IMHO, that means that `isinstance(CustomError(..), Exception)` should be `True`.
That's the case, because the base case of `RuntimeError` is `Exception`.
There's a nice hierarchy tree at the end of the page you linked.
AFAIK, it's a usual practice to subclass `RuntimeError` or any other python built-in exception.
e.g.: `RCLError` subclass from `RuntimeError`.

@wjwwood what do you think?

Reply:
Correct, you can, and where appropriate, you should inherit from other exception types. They just mean you should raise a completely custom class that does not inherit from exception at all, directly or indirectly.

Reply:
I will move the definition of this exception to `_rclpy.c` as suggested at https://github.com/ros2/rclpy/pull/459#discussion_r397849914 , in which case this exception will inherit from `RCLError` instead of `RuntimeError`.